src/arith/iter_affine_map.cc

```diff
@@ -898,6 +898,11 @@ class IterMapRewriter : public ExprMutator {
   PrimExpr SplitFloorModConst(IterSplitExpr lhs, PrimExpr base, PrimExpr rhs);
 
   static void AddToLhs(IterSumExprNode* lhs, IterSplitExpr rhs, int sign) {
+    if (sign < 0 && is_const_int(rhs->extent, 2)) {
```

Comment:
Could we add a cover case for the codepath?

Reply:
Can do, and added!  This was a case that was caught by other unit tests, since the additional `RewriteSimplifier` rules prevented `DetectIterMap` from recognizing some patterns after they had been simplfied, but a specific unit test for this case is better than needing to track it down later.

Reply:
NOTE: this actually is a bug, see https://github.com/apache/tvm/pull/14571

This makes me think that if other rules leads to the regression. Please do check.